mesonbuild/interpreter.py

```diff
@@ -2211,7 +2218,16 @@ def func_subproject(self, nodes, args, kwargs):
         dirname = args[0]
         return self.do_subproject(dirname, kwargs)
 
+    def disabled_subproject(self, dirname):
+        self.subprojects[dirname] = Disabler()
+        return self.subprojects[dirname]
```

Comment:
Other APIs with a "required" kwarg does not return a Disabler object because it has special behaviour that the user needs to handle correctly. I think you should return a SubprojectHolder that has a found() method returning false and get_variable_method() raising exception.

See PR #3386 to add a way for those methods to return a disabler.

Reply:
I like the idea of returnin a Disabler, but let's be consistent indeed.